security_simulation/attendee.py

```diff
@@ -105,18 +106,25 @@ def find_checkpoint(self, checkpoints):
         checkpoint_rankings = checkpoint_ratios + checkpoint_line_len
         min_index = N.argmin(checkpoint_rankings)
         # found the target checkpoint, set that as the target_checkpoint
-        self.checkpoint_target = checkpoints[min_index]
-        self._calc_checkpoint_arrival(checkpoint_distances[min_index])
-        self._set_checkpoint_vector(self.checkpoint_target.get_location())
+        checkpoint_candidate = checkpoints[min_index]
+        if self.checkpoint_target is None or self.checkpoint_target is not checkpoint_candidate:
+            if self.checkpoint_target is not None:
+                print("Attendee:", self.attendee_id, "has changed checkpoint target from:",\
```

Comment:
add a flag to turn this print off, i imagine it might slow things down considerably